src/stream/src/executor/aggregation/agg_state.rs

```diff
@@ -45,16 +49,24 @@ impl<S: StateStore> Debug for AggState<S> {
 pub const ROW_COUNT_COLUMN: usize = 0;
 
 impl<S: StateStore> AggState<S> {
-    pub fn row_count(&mut self) -> i64 {
-        if let ManagedStateImpl::Value(count_state) = &self.managed_states[ROW_COUNT_COLUMN] {
-            count_state.get_output().map(|x| *x.as_int64()).unwrap_or(0)
-        } else {
-            unreachable!("streamAgg's count state must be a value state");
+    pub fn new(
+        group_key: Option<Row>,
+        managed_states: Vec<ManagedStateImpl<S>>,
+        prev_outputs: Option<Vec<Datum>>,
+    ) -> Self {
+        Self {
+            group_key,
+            managed_states,
+            prev_outputs,
         }
     }
 
+    pub fn managed_states(&mut self) -> &mut Vec<ManagedStateImpl<S>> {
```

Comment:
```suggestion
    pub fn managed_states(&mut self) -> &mut [ManagedStateImpl<S>] {
```
May return a slice as the vec never grows.